tests/test_transfer.py

```diff
@@ -116,21 +116,13 @@ def test_transfer_complex_4():
 def test_transfer_complex_5():
     # check a generic decomposition
     with open(TEST_DATA_DIR / 'u_random_8', "r") as f:
-        M = Matrix(f)
-        def ub(idx):
-            return (Circuit(2)
-                    // comp.BS()
-                    // (0, comp.PS(phi=P("φ_a%d" % idx)))
-                    // comp.BS()
-                    // (0, comp.PS(phi=P("φ_b%d" % idx))))
-        C1 = Circuit.decomposition(M, ub(0), shape=InterferometerShape.TRIANGLE)
-    C2 = GenericInterferometer(8, ub, shape=InterferometerShape.TRIANGLE)
-    C2.transfer_from(C1)
-    def ub_varbs(idx):
-        return (Circuit(2)
-                // comp.BS(theta=P("theta%d" % (2*idx)))
-                // (0, comp.PS(phi=0))
-                // comp.BS(theta=P("theta%d" % (2*idx+1)))
-                // (0, comp.PS(phi=0)))
-    C3 = GenericInterferometer(8, ub_varbs, shape=InterferometerShape.TRIANGLE)
-    C3.transfer_from(C1, force=True)
+        m = Matrix(f)
+        def ub():
+            return catalog["mzi phase last"].build_circuit() // comp.Barrier(2, visible=False)
+        c1 = Circuit.decomposition(m, ub(), shape=InterferometerShape.TRIANGLE)
+    c2 = GenericInterferometer(8, catalog["mzi phase last"].generate, shape=InterferometerShape.TRIANGLE)
+    c2.transfer_from(c1)
+    def ub_varbs(idx: int):
+        return catalog["mzi phase last"].build_circuit(theta_a=f"theta{2*idx}", theta_b=f"theta{2*idx+1}", phi_a=0, phi_b=0)
+    c3 = GenericInterferometer(8, ub_varbs, shape=InterferometerShape.TRIANGLE)
+    c3.transfer_from(c1, force=True)
```

Comment:
This test seems to perform a generic decomposition and then transfer parameters. There are no assert statements or other checks.